### ISSUE TITLE
Missing ionization potential subtraction in  the Thomas-Fermi ionization model

### ISSUE THREAD

**Issue by pordyna** [ComputationalRadiationPhysics/picongpu#4485]
@Lingen1218 and Long Yang noticed that we are not subtracting the ionization potentials from the ion background in the Thomas-Fermi implementation of the collisional ionization. We do subtract the new electron momentum from the ion, but we should also conserve the electron binding energy. I had a quick look at the code and I also couldn't find it anywhere in `PIConGPU`. 

In case this is in fact missing, one possible way of doing it would be to sum over ionization potentials of all freed electrons in a cell while creating particles, and spreading it afterward among all ions in the cell in a separate kernel.

@BrianMarre It seems to be, again, your area of expertise.

**Comment by BrianMarre** [comment]
Yes you are correct, we do not take account for the ionization energies anywhere as far as I know. Marco just started looking into that with the ionization current for field ionization.
But wouldn't it make more sense to extract the energy from both electrons and ions?
Something along the lines of sum up all the ionization energies and then do a binary collision step and extract the average energy from all pairs' momenta by random inelastic collisions?

**Comment by PrometheusPi** [comment]
@pordyna and @BrianMarre  Are there plans to implement energy conservation for Thomas Fermi soon?

**Comment by BrianMarre** [comment]
will discuss with @HighIander
